lib/rack/user_agent/detector.rb

```diff
@@ -32,10 +36,18 @@ def from_android_tablet?
         os == "Android" && !android_mobile?
       end
 
+      def from_android_os?
+        from_android? || from_android_tablet?
+      end
+
       def from_windows_phone?
         os == "Windows Phone OS"
       end
 
+      def from_windows_phone_os?
+        from_windows_phone?
```

Comment:
I think it'd be good to merge `#from_ios?` and `#from_android_os?` since you can easily write the combinations. But why do you want `#from_windows_phone_os?`?